pl_examples/utils/loaders.py

```diff
@@ -0,0 +1,77 @@
+from threading import Thread
+from queue import Queue
+
+import torch
+from torch import Tensor
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.utils.data import DataLoader
+
+
+class AsynchronousLoader(DataLoader):
```

Comment:
One question on this: Let's say I use a custom data loader. Can I also use this async loader? I think I can't since the logic I implemented myself would be missing, right? Is there a possibility to just wrap the already specified loader instead of replacing it?

Reply:
Yes, it is be possible. The only issue I've seen so far arises when the custom data loader is infinite (or of an unknown length). Then the termination of the loading thread becomes more complicated

Reply:
Let's just say, that the user can give an explicit length argument (defaults to None) and include a check, whether this argument was set or the loader contains a length